lib/service/connect.go

```diff
@@ -160,6 +160,15 @@ func (process *TeleportProcess) connect(role types.SystemRole) (conn *Connector,
 		process.log.Infof("Connecting to the cluster %v with TLS client certificate.", identity.ClusterName)
 		clt, err := process.newClient(process.Config.AuthServers, identity)
 		if err != nil {
+			// In the event that a user is attempting to connect a machine to
+			// a different cluster it will give a cryptic warning about an
+			// unknown certificate authority. Unfortunately we cannot intercept
+			// this error as it comes from the http package before a request is
+			// made. So provide a more user friendly error as a hint of what
+			// they can do to resolve the issue.
+			if strings.Contains(err.Error(), "certificate signed by unknown authority") {
+				process.log.Error("Was this node already registered to a different cluster? To join this node to a new cluster, remove `/var/lib/teleport` and try again")
```

Comment:
@zmb3 what do you think of 
```suggestion
				process.log.Error("Was this node already registered to a different cluster? To join this node to a new cluster remove its data directory and try again (default location: `/var/lib/teleport`).")
```

Reply:
You have access to the config, so `process.Config.DataDir` will give you the exact directory.

Reply:
Brilliant thanks!